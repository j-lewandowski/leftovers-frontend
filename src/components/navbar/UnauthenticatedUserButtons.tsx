import CategoriesButton from '../buttons/CategoriesButton';
import SigninButton from '../buttons/SigninButton';
import SignupButton from '../buttons/SignupButton';

const UnauthenticatedUserButtons = () => {
  return (
    <>
      <CategoriesButton />
      <SigninButton />
      <SignupButton />
    </>
  );
};

export default UnauthenticatedUserButtons;
