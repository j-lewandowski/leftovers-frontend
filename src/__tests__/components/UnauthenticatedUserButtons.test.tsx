import { render } from '@testing-library/react';
import { describe, test, expect } from 'vitest';
import UnauthenticatedUserButtons from '../../components/navbar/UnauthenticatedUserButtons';

describe('UnauthenticatedUserButtons', () => {
  test('should render unauthenticated user buttons', () => {
    const result = render(<UnauthenticatedUserButtons />);
    expect(result).toMatchSnapshot();
  });
});
