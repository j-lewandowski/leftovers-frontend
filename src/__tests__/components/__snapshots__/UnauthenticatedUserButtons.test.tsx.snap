// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UnauthenticatedUserButtons > should render unauthenticated user buttons 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorSecondary MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorSecondary css-1d3f8j8-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Recipes
        </p>
        <span
          class="MuiButton-icon MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-16tdoj9-MuiSvgIcon-root"
            data-testid="ExpandMoreIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Log in
        </p>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-sghohy-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Sign up
        </p>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorSecondary MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorSecondary css-1d3f8j8-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        Recipes
      </p>
      <span
        class="MuiButton-icon MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-16tdoj9-MuiSvgIcon-root"
          data-testid="ExpandMoreIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        Log in
      </p>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-sghohy-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        Sign up
      </p>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
